test/ably/restsetup.py

```diff
@@ -17,26 +16,22 @@
     app_spec_local = json.loads(f.read())
 
 tls = (os.environ.get('ABLY_TLS') or "true").lower() == "true"
-host = os.environ.get('ABLY_HOST')
+host = os.environ.get('ABLY_HOST', 'sandbox-rest.ably.io')
+environment = os.environ.get('ABLY_ENV')
```

Comment:
If we're removing support for `ABLY_HOST` this can be a lot simpler.  Everything is based on the given environment, if any.

```
environment = os.environ.get('ABLY_ENV', 'sandbox')
host = (environment == 'production) ? 'rest.ably.io' : environment + '_rest.ably.io'
port = (environment == 'local') ? 8080 : 80
tls_port = (environment == 'local') ? 8081 : 443
```

Reply:
Could we handle this in a different issue? Because I'm having errors using `environment`.

Everything else in this PR is green now, and this PR has grown quite a lot with all the test suite improvements.

Reply:
> Could we handle this in a different issue?

Yes of course

Reply:
cool, then could you approve the PR so I can merge it?

Reply:
See #123 for the new issue